examples/api/lib/widgets/interactive_viewer/interactive_viewer.builder.0.dart

```diff
@@ -150,38 +98,42 @@ typedef _CellBuilder = Widget Function(
 
 class _TableBuilder extends StatelessWidget {
   const _TableBuilder({
-    required this.rowCount,
-    required this.columnCount,
     required this.cellWidth,
+    required this.cellHeight,
+    required this.viewport,
     required this.builder,
-  })  : assert(rowCount > 0),
-        assert(columnCount > 0);
+  });
 
-  final int rowCount;
-  final int columnCount;
   final double cellWidth;
+  final double cellHeight;
+  final Rect viewport;
   final _CellBuilder builder;
 
   @override
   Widget build(BuildContext context) {
-    return Table(
-      // ignore: prefer_const_literals_to_create_immutables
-      columnWidths: <int, TableColumnWidth>{
-        for (int column = 0; column < columnCount; column++)
-          column: FixedColumnWidth(cellWidth),
-      },
-      // ignore: prefer_const_literals_to_create_immutables
-      children: <TableRow>[
-        for (int row = 0; row < rowCount; row++)
-          // ignore: prefer_const_constructors
-          TableRow(
-            // ignore: prefer_const_literals_to_create_immutables
-            children: <Widget>[
-              for (int column = 0; column < columnCount; column++)
-                builder(context, row, column),
-            ],
-          ),
-      ],
-    );
+    final int firstRow = (viewport.top / cellHeight).floor();
+    final int lastRow = (viewport.bottom / cellHeight).ceil();
+    final int firstCol = (viewport.left / cellWidth).floor();
+    final int lastCol = (viewport.right / cellWidth).ceil();
+
+    // This will create and render exactly (lastRow - firstRow) * (lastCol - firstCol) cells
+
+    return SizedBox(
+        // Stack needs constraints, even though we then Clip.none outside of them.
+        // InteractiveViewer.builder always sets constrained to false, giving infinite constraints to the child.
+        // See: https://master-api.flutter.dev/flutter/widgets/InteractiveViewer/constrained.html
+        width: 1,
+        height: 1,
+        child: Stack(
+          clipBehavior: Clip.none,
+          children: <Widget>[
+            for (int row = firstRow; row < lastRow; row++)
+              for (int col = firstCol; col < lastCol; col++)
+                Positioned(
+                    left: col * cellWidth,
+                    top: row * cellHeight,
+                    child: builder(context, row, col)),
+          ],
+        ));
```

Comment:
Add a couple of commas and reformat for better formatting.

```suggestion
                child: builder(context, row, col),
              ),
        ],
      ),
    );
```